src/load/azext_load/data_plane/load_test/custom.py

```
@@ -136,8 +146,40 @@ def create_test(
     upload_files_helper(
         client, test_id, yaml, test_plan, load_test_config_file, not custom_no_wait, evaluated_test_type
     )
-    response = client.get_test(test_id)
     logger.info("Upload files to test %s has completed", test_id)
+    if app_components is not None and len(app_components) > 0:
+        # only get and patch the app components if its present in the yaml.
+        try:
+            app_components_existing = client.get_app_components(test_id)
+        except Exception:
+            app_components_existing = {"components": {}}
+            pass
+        app_components_merged = merge_existing_app_components(
+            app_components, app_components_existing.get("components", {})
+        )
+        app_component_response = client.create_or_update_app_components(
+            test_id=test_id, body={"testId": test_id, "components": app_components_merged}
+        )
+        logger.warning(
+            "Added app components for test ID: %s and response obj is %s", test_id, app_component_response
+        )
+    if server_metrics is not None and len(server_metrics) > 0:
+        # only get and patch the app components if its present in the yaml.
+        try:
+            server_metrics_existing = client.get_server_metrics_config(test_id)
+        except Exception:
+            server_metrics_existing = {"metrics": {}}
```

what if the exception is some internal error but there is supposed to be existing data, it will get overridden?

---

we will just ignore the error and consider it as patch call now.

---

ADO follows